module/VuFind/src/VuFind/ILS/Driver/NoILS.php

```diff
@@ -269,8 +269,9 @@ protected function getFormattedMarcDetails($recordDriver, $configSection)
             // ID prefix, strip it off!
             $idPrefix = $this->getIdPrefix();
             if (
-                isset($result[0]['id']) && strlen($idPrefix)
-                && $idPrefix === substr($result[0]['id'], 0, strlen($idPrefix))
+                isset($result[0]['id'])
+                && '' !== $idPrefix
```

Comment:
Do we need this check? If the string is empty, I would expect str_starts_with to return false anyway.

Reply:
Yes, it's needed. "[All strings start with the empty string](https://www.php.net/manual/en/function.str-starts-with.php)"...

Reply:
Ah, sorry, I misread that and thought you were checking for an empty haystack rather than an empty needle! :-)